src/Servers/Kestrel/test/Interop.FunctionalTests/HttpClientHttp2InteropTests.cs

```diff
@@ -0,0 +1,594 @@
+                            allRequestsReceived.SetResult(0);
+                        }
+                        await allRequestsReceived.Task;
+                        await context.Request.BodyReader.CopyToAsync(context.Response.BodyWriter);
+                    }));
+                });
+            using var host = await hostBuilder.StartAsync();
+
+            var url = $"{scheme}://127.0.0.1:{host.GetPort().ToString(CultureInfo.InvariantCulture)}/";
+
+            using var client = CreateClient();
+            client.DefaultRequestHeaders.ExpectContinue = true;
+
+            var requestTasks = new List<Task>(requestCount);
+            for (var i = 0; i < requestCount; i++)
+            {
+                requestTasks.Add(RunRequest(url));
+            }
+
+            async Task RunRequest(string url)
+            {
+                using var request = new HttpRequestMessage(HttpMethod.Post, url)
+                {
+                    Version = HttpVersion.Version20,
+                    Content = new BulkContent()
+                };
+                using var response = await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+
+                Assert.Equal(HttpVersion.Version20, response.Version);
+                await BulkContent.VerifyContent(await response.Content.ReadAsStreamAsync());
+            };
+
+            await Task.WhenAll(requestTasks);
+            await host.StopAsync();
+        }
+
+        private class BulkContent : HttpContent
+        {
+            private static readonly byte[] Content;
+            private static readonly int Repititions = 200;
+
+            static BulkContent()
+            {
+                Content = new byte[999]; // Intentionally not matching normal memory page sizes to ensure we stress boundaries.
+                for (var i = 0; i < Content.Length; i++)
+                {
+                    Content[i] = (byte)i;
+                }
+            }
+
+            protected override async Task SerializeToStreamAsync(Stream stream, TransportContext context)
+            {
+                for (var i = 0; i < Repititions; i++)
+                {
+                    using (var timer = new CancellationTokenSource(TimeSpan.FromSeconds(50000)))
+                    {
+                        await stream.WriteAsync(Content, 0, Content.Length, timer.Token);
+                    }
+                    await Task.Yield(); // Intermix writes
+                }
+            }
+
+            protected override bool TryComputeLength(out long length)
+            {
+                length = 0;
+                return false;
+            }
+
+            public static async Task VerifyContent(Stream stream)
+            {
+                byte[] buffer = new byte[1024];
+                var totalRead = 0;
+                var patternOffset = 0;
+                int read = 0;
+                using (var timer = new CancellationTokenSource(TimeSpan.FromSeconds(5000)))
```

Comment:
This timeout seems too large. Maybe just 60 seconds?

Reply:
😁 Leftover from debugging.